django/db/models/fields/mixins.py

```diff
@@ -12,6 +12,12 @@ def get_cached_value(self, instance, default=NOT_PROVIDED):
         try:
             return instance._state.fields_cache[cache_name]
         except KeyError:
+            ancestor_link = instance._meta.get_ancestor_link(self.model)
```

Comment:
I think a comment before this block of code would be useful as the circumstances of when it runs isn't self-evident to me.